lib/dashing-api.rb

```diff
@@ -1,193 +1,187 @@
-require 'helperFunctions'
-
-functions = HelperFunctions.new
-
-#Dashing endpoint to get the list of dashboards
-get '/dashboard/list' do
-  content_type :json
-  result = %x`ls dashboards/`
-  return result
-end
-
-# Dashing endpoint to get the current data of a tile
-get '/widgets/:id.json' do
-  content_type :json
-  if data = settings.history[params[:id]]
-     data.split[1]
-  end
-end  
-
-# Check if a nagios host has a job script
-get '/hosts/:id' do
-   content_type :json
-   hostName = params[:id]
-   if settings.history[hostName]
-      "Nagios host has a job script"
-      { :host => hostName, :status => '200 OK' }.to_json
-   else
-      { :host => hostName, :status => '404 Not Found' }.to_json
-   end
-end
-
-# Get a list of all existing widgets
-get '/widgets/' do
-  content_type :json
-  %x`ls widgets/`
-end
-
-
-# Get a list of a existing dashboards
-get '/dashboards/' do
-  content_type :json
-  output = %x`ls dashboards/*.erb | xargs -n1 basename`
-  output.reverse
-  output.split('/n')
-end
-
-
-# Check if a dashboard exists
-get '/dashboards/:id' do
-  dashboard = params[:id]    
-  if functions.dashboardExists(dashboardName)
-     { :dashboard => dashboard, :status => '200 OK' }.to_json
-  else
-     { :dashboard => dashboard, :status => '404 Not Found' }.to_json
-  end
-end
-
+require 'json'
+require './helperFunctions'
```

Comment:
This should just work as `require 'helperFunctions'` according to: http://stackoverflow.com/questions/1050749/including-a-ruby-class-from-a-separate-file